py/dml/template.py

```diff
@@ -144,6 +156,88 @@ def traits(self):
             return Set().union(
                 *[sup.traits() for (_, sup) in self.spec.templates])
 
+    @functools.cache
+    def get_potential_method_impl_details(self, method_name):
```

Comment:
I know you hate type annotations, but they are a quite nice way to document return tuples.

Reply:
errrgggghhhh....

i guess. I'm fine with type annotations as long as we have no illusions that they are anything more _but_ documentation

Reply:
Done.